amigo-lib/src/main/java/me/ele/amigo/ActivityStub.java

```diff
@@ -0,0 +1,247 @@
+package me.ele.amigo;
+
+import android.app.Activity;
+import android.content.pm.ActivityInfo;
+import android.util.Log;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import static android.content.pm.ActivityInfo.LAUNCH_SINGLE_INSTANCE;
+import static android.content.pm.ActivityInfo.LAUNCH_SINGLE_TASK;
+import static android.content.pm.ActivityInfo.LAUNCH_SINGLE_TOP;
+
+public abstract class ActivityStub extends Activity {
+
+    private static final List<Class<? extends ActivityStub>> SINGLE_TOP_STUBS = Arrays.asList(
+            SingleTopStub1.class,
+            SingleTopStub2.class,
+            SingleTopStub3.class,
+            SingleTopStub4.class,
+            SingleTopStub5.class,
+            SingleTopStub6.class,
+            SingleTopStub7.class,
+            SingleTopStub8.class);
+
+    private static final List<Class<? extends ActivityStub>> SINGLE_TASK_STUBS = Arrays.asList(
+            SingleTaskStub1.class,
+            SingleTaskStub2.class,
+            SingleTaskStub3.class,
+            SingleTaskStub4.class,
+            SingleTaskStub5.class,
+            SingleTaskStub6.class,
+            SingleTaskStub7.class,
+            SingleTaskStub8.class);
+
+    private static final List<Class<? extends ActivityStub>> SINGLE_INSTANCE_STUBS = Arrays.asList(
+            SingleInstanceStub1.class,
+            SingleInstanceStub1.class,
+            SingleInstanceStub2.class,
+            SingleInstanceStub3.class,
+            SingleInstanceStub4.class,
+            SingleInstanceStub5.class,
+            SingleInstanceStub6.class,
+            SingleInstanceStub7.class,
+            SingleInstanceStub8.class);
+
+    // [stub activity name : target activity instance]
+    private static final Map<String, ActivityRecord> usedSingleTopStubs = new LinkedHashMap<>(SINGLE_TOP_STUBS.size(), 0.75f, true);
+    private static final Map<String, ActivityRecord> usedSingleTaskStubs = new LinkedHashMap<>(SINGLE_TASK_STUBS.size(), 0.75f, true);
+    private static final Map<String, ActivityRecord> usedSingleInstanceStubs = new LinkedHashMap<>(SINGLE_INSTANCE_STUBS.size(), 0.75f, true);
+
+    private static Class findUnusedActivityStub(String componentName, Map<String, ActivityRecord> recordMap, List<Class<? extends ActivityStub>> stubs) {
+        Log.d("stub", "findUnusedActivityStub: component = " + componentName);
+        for (Class stub : stubs) {
+            ActivityRecord record = recordMap.get(stub.getName());
+            if (record != null && record.activityClazzName.equals(componentName)) {
```

Comment:
here confuse me, this method named as `findUnusedActivityStub`, but seems it not just find the unused activity stub, it will return the currently using stub for the component, right? Would you rename the method or do some tiny refactor?